sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1528,6 +1528,23 @@ abstract class DynamicPartitionPruningSuiteBase
       }
     }
   }
+
+  test("SPARK-37995: PlanAdaptiveDynamicPruningFilters should use prepareExecutedPlan " +
```

Comment:
Let's put the test in `DynamicPartitionPruningV1SuiteAEOn`, instead of running it 4 times with the same setting.